src/Lecture2.hs

```diff
@@ -268,7 +362,19 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval _ (Lit x) = Right x
+eval [] (Var varName) = Left (VariableNotFound varName)
```

Comment:
It looks like a possible optimization 🤔 
But I think this case is redundant as `lookup` on an empty list will always return `Nothing` so you don't need to handle this separately